### ISSUE TITLE
@@dancing.utils/tick_and_dispatch CSRF protection fail after plone4.csrffixes patch

### ISSUE THREAD

**Issue by ivanteoh** [collective/collective.dancing#24]
@@dancing.utils/tick_and_dispatch can be called from different way. 

```
plone.protect aborting transaction due to no CSRF protection on url http://localhost:18881/Plone/@@dancing.utils/tick_and_dispatch
```

After plone4.csrffixes is patched, ways seem failed:
* directly call `@@dancing.utils/tick_and_dispatch`
* call from clock-server

It seems okay when it is called from collective.cron.

It failed because it happen to be writes-on-read. When this url is called, property message_sent will calculated in MessageQueues, which is a PersistentDict.

What is the best solution for this issue?
* don't fix. Just used collective.cron.
* don't fix. Use `HTTP_REFERER or HTTP_ORIGIN` when calling `@@dancing.utils/tick_and_dispatch`
* fix. White list this url.

Form this article, http://devblog.4teamwork.ch/blog/2015/10/12/debugging-csrf-protection-false-positives-in-plone/, we can white list the method.

```
from plone.protect.auto import safeWrite

safeWrite(myobj, request)
````

**Comment by dnouri** [comment]
Whitelisting sounds like a good option.

**Comment by ivanteoh** [comment]
Whitelist is good. Unfortunately, there are three obj that need to whitelist, refer https://gist.github.com/ivanteoh/83e2dfc3f6510b7deacc. I guess the last resort is completely disabling CSRF protection for the requests. Any comments?

**Comment by frisi** [comment]
fixed by PR #25
